Upgrade to chokidar 4.0.3

### Describe the problem

The latest chokidar causes type-checking to fail due to https://github.com/paulmillr/chokidar/pull/1381

### Describe the proposed solution

This line will probably need to be tweaked by someone who knows TypeScript:

https://github.com/sveltejs/kit/blob/01f001bd861e399027a580c9402b7b6206de3d24/packages/package/src/index.js#L81

```
src/index.js:81:61 - error TS2345: Argument of type '(value: any) => void' is not assignable to parameter of type '() => void'.
  Target signature provides too few arguments. Expected 1 or more, but got 0.

81  const ready = new Promise((resolve) => watcher.on('ready', resolve));
```

### Alternatives considered

_No response_

### Importance

nice to have

### Additional Information

_No response_

--- a/.changeset/neat-mirrors-rush.md
+++ b/.changeset/neat-mirrors-rush.md
@@ -1,0 +1,5 @@
+---
+"@sveltejs/package": patch
+---
+
+chore: update to chokidar 4.0.3

--- a/packages/package/package.json
+++ b/packages/package/package.json
@@ -20,7 +20,7 @@
 	"homepage": "https://svelte.dev",
 	"type": "module",
 	"dependencies": {
-		"chokidar": "^4.0.0",
+		"chokidar": "^4.0.3",
 		"kleur": "^4.1.5",
 		"sade": "^1.8.1",
 		"semver": "^7.5.4",

--- a/packages/package/src/index.js
+++ b/packages/package/src/index.js
@@ -78,6 +78,7 @@ export async function watch(options) {
 	let timeout;
 
 	const watcher = chokidar.watch(input, { ignoreInitial: true });
+	/** @type {Promise<void>} */
 	const ready = new Promise((resolve) => watcher.on('ready', resolve));
 
 	watcher.on('all', (type, filepath) => {

